client/consensus/aura/src/collators/basic.rs

```diff
@@ -0,0 +1,199 @@
+		+ Send
+		+ Sync
+		+ 'static,
+	Client::Api: AuraApi<Block, P::Public> + CollectCollationInfo<Block>,
+	RClient: RelayChainInterface,
+	CIDP: CreateInherentDataProviders<Block, ()> + 'static,
+	BI: BlockImport<Block> + ParachainBlockImportMarker + Send + Sync + 'static,
+	SO: SyncOracle + Send + Sync + Clone + 'static,
+	Proposer: ProposerInterface<Block, Transaction = BI::Transaction>,
+	Proposer::Transaction: Sync,
+	CS: CollatorServiceInterface<Block>,
+	P: Pair + Send + Sync,
+	P::Public: AppPublic + Hash + Member + Encode + Decode,
+	P::Signature: TryFrom<Vec<u8>> + Hash + Member + Encode + Decode,
+{
+	let mut collation_requests = cumulus_client_collator::relay_chain_driven::init(
+		params.key,
+		params.para_id,
+		params.overseer_handle,
+	)
+	.await;
+
+	let mut collator = {
+		let params = collator_util::Params {
+			create_inherent_data_providers: params.create_inherent_data_providers,
+			block_import: params.block_import,
+			relay_client: params.relay_client.clone(),
+			keystore: params.keystore.clone(),
+			para_id: params.para_id,
+			proposer: params.proposer,
+			collator_service: params.collator_service,
+		};
+
+		collator_util::Collator::<Block, P, _, _, _, _, _>::new(params)
+	};
+
+	while let Some(request) = collation_requests.next().await {
+		macro_rules! reject_with_error {
+			($err:expr) => {{
+				request.complete(None);
+				tracing::error!(target: crate::LOG_TARGET, err = ?{ $err });
+				continue;
+			}};
+		}
+
+		macro_rules! try_request {
+			($x:expr) => {{
+				match $x {
+					Ok(x) => x,
+					Err(e) => reject_with_error!(e),
+				}
+			}};
+		}
+
+		let validation_data = request.persisted_validation_data();
+
+		let parent_header =
+			try_request!(Block::Header::decode(&mut &validation_data.parent_head.0[..]));
+
+		let parent_hash = parent_header.hash();
+
+		if !collator.collator_service().check_block_status(parent_hash, &parent_header) {
+			continue
+		}
+
+		let relay_parent_header = match params.relay_client.header(*request.relay_parent()).await {
+			Err(e) => reject_with_error!(e),
+			Ok(None) => continue, // sanity: would be inconsistent to get `None` here
+			Ok(Some(h)) => h,
+		};
+
+		let claim = match collator_util::claim_slot::<_, _, P>(
+			&*params.para_client,
+			parent_hash,
+			&relay_parent_header,
+			params.slot_duration,
+			params.relay_chain_slot_duration,
+			&params.keystore,
+		)
+		.await
+		{
+			Ok(None) => continue,
+			Ok(Some(c)) => c,
+			Err(e) => reject_with_error!(e),
+		};
+
+		let (parachain_inherent_data, other_inherent_data) = try_request!(
+			collator
+				.create_inherent_data(
+					*request.relay_parent(),
+					&validation_data,
+					parent_hash,
+					claim.timestamp(),
+				)
+				.await
+		);
+
+		let (collation, _, post_hash) = try_request!(
+			collator
+				.collate(
+					&parent_header,
+					&claim,
+					None,
+					(parachain_inherent_data, other_inherent_data),
+					// TODO [https://github.com/paritytech/cumulus/issues/2439]
```

Comment:
Perhaps we can get rid of this TODO. Seems that you handled authoring_duration in the collator params of lookahead.rs

Reply:
Unfortunately, that doesn't really work for the 'basic' collator as the time limitations are very strict (and this PR just maintains behavior as this code is becoming obsolete)